bootstrap.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+"""Bootstrap script for emscripten developers / git users.
+
+After checking out emscripten there are certain steps that need to be
+taken before it can be used.  This script enumerates and automates
+these steps and is able to run just the steps that are needed based
+on the timestamps of various input files (kind of like a dumb version
+of a Makefile).
+"""
+import argparse
+import os
+import shutil
+import sys
+
+__rootdir__ = os.path.dirname(os.path.abspath(__file__))
+sys.path.insert(0, __rootdir__)
+
+STAMP_DIR = os.path.join(__rootdir__, 'out')
+
+from tools import shared, utils
+
+actions = [
+  ('npm packages', 'package.json', [shutil.which('npm'), 'ci']),
+  # TODO(sbc): Remove the checked in entry point files and have them
+  # built on demand by this step.
+  ('create entry points', 'tools/maint/create_entry_points.py', [sys.executable, 'tools/maint/create_entry_points.py']),
+  ('git submodules', 'test/third_party/posixtestsuite/', [shutil.which('git'), 'submodule', 'update', '--init']),
```

Comment:
Requiring all Emscripten git users to clone posixtestsuite as well is maybe a bit excessive. It would be better to have the test runner complain only if attempting to run `test_posixtest.py` and posixtestsuite is not cloned - 99% of users will never run that suite.

Also given that posixtestsuite is GPL, it would be good to keep its integration clear from the main repo to make it obvious that Emscripten is not a derivative work building on top of posixtestsuite.

Reply:
I guess that question is how many git user of emscripten are there and what subset don't want the submodules?  As it happens the submodules we have today are not essential to emscripten's basic operation so I guess we could consider removing this step.. but if we ever did have an essential submodule we would need to add it back.

Reply:
It could probably be not a submodule but rather an explicit `git clone` done only by `test_posixtest.py`?

Reply:
Yes, that might help with the GPL coupling issue I guess. 

On the other hand we might want to wait until sombody actually raises a issue about that.  Since this does is not part of `make install` (or `tools/install.py`) is should not show up in any distributions or installations of emscripten (e.g. emsdk).